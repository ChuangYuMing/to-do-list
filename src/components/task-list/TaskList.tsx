import { Task } from "@/types";
import TaskItem from "../task-item/TaskItem";
import styles from "./TaskList.module.scss";

type TaskListProps = {
  filterTasks: Task[];
  updateTask: (taskId: number, payload: Partial<Task>) => void;
  deleteTask: (taskId: number) => void;
};

export default function TaskList({
  filterTasks,
  updateTask,
  deleteTask,
}: TaskListProps) {
  const taskList = filterTasks.map((task) => (
    <TaskItem
      key={task.createdAt}
      task={task}
      updateTask={updateTask}
      deleteTask={deleteTask}
    />
  ));

  return <div className={styles["task-list-wrap"]}>{taskList}</div>;
}
