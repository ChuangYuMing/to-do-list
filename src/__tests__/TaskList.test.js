import React from "react";
import { render } from "@testing-library/react";
import TaskList from "../components/task-list/TaskList";

it("renders correctly", () => {
  const mockFilterTasks = [
    {
      createdAt: 1718944797228,
      id: 1,
      description: "Task 1",
      completed: false,
    },
    { createdAt: 1718924797228, id: 2, description: "Task 2", completed: true },
  ];

  const { asFragment } = render(
    <TaskList
      filterTasks={mockFilterTasks}
      updateTask={() => {}}
      deleteTask={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
