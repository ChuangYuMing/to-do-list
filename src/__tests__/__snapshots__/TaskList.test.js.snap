// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="task-list-wrap"
  >
    <div
      class="task-item"
    >
      <div
        class="task-content"
      >
        <input
          class="task-checkbox"
          id=":r0:"
          type="checkbox"
        />
        <label
          class="checkbox-label"
          for=":r0:"
        />
        <span
          class="task-description"
          role="description"
        >
          Task 1
        </span>
      </div>
      <div
        class="task-actions"
      >
        <span
          class="task-action"
          role="deleteTask"
        >
          Delete
        </span>
      </div>
      <span
        class="created-date"
      >
        Jun 21 2024, 12:39:57
      </span>
    </div>
    <div
      class="task-item"
    >
      <div
        class="task-content"
      >
        <input
          checked=""
          class="task-checkbox"
          id=":r1:"
          type="checkbox"
        />
        <label
          class="checkbox-label"
          for=":r1:"
        />
        <span
          class="task-description"
          role="description"
        >
          Task 2
        </span>
      </div>
      <div
        class="task-actions"
      >
        <span
          class="task-action"
          role="deleteTask"
        >
          Delete
        </span>
      </div>
      <span
        class="created-date"
      >
        Jun 21 2024, 07:06:37
      </span>
    </div>
  </div>
</DocumentFragment>
`;
